src/coprocessor/statistics/analyze.rs

```diff
@@ -130,9 +142,38 @@ impl<S: Snapshot> AnalyzeContext<S> {
                     cms.insert(&data);
                 }
             }
+            if stats_version == VERSION2 {
+                let vec_data = data.to_vec();
+                if cur_v.1 == vec_data {
+                    cur_v.0 += 1;
+                } else {
+                    if cur_v.0 > 0 {
+                        topn_heap.push(Reverse(cur_v.clone()));
+                    }
+                    if topn_heap.len() > top_n_size {
+                        topn_heap.pop();
+                    }
+                    cur_v = (1, vec_data);
+                }
+            }
             hist.append(&data);
         }
 
+        if stats_version == VERSION2 {
+            if cur_v.0 > 0 {
+                topn_heap.push(Reverse(cur_v.clone()));
+                if topn_heap.len() > top_n_size {
+                    topn_heap.pop();
+                }
+            }
+            if let Some(c) = cms.as_mut() {
+                for heap_data in topn_heap {
+                    c.sub(&(heap_data.0).1, (heap_data.0).0 as u32);
+                    c.push_to_top_n(&(heap_data.0).1, (heap_data.0).0 as u64);
```

Comment:
After addressing https://github.com/tikv/tikv/pull/8177/files#r539287646, `heap_data.0.1` can be passed to this function directly to avoid allocation.